### DISCUSSION TITLE
schema help

### DISCUSSION THREAD

**Discussion by jdev04** [prisma/prisma#12461] [Q&A]
I need prisma help, Im trying to make it so the uploader of a post can add contributors that that are linked to other users and allow them to be showed on their profile. If they dont have a profile, the uploader can opt to use another link. How do i accomplish this?

Edit: I still have more to do other than this^
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  referentialIntegrity = "prisma"
  shadowDatabaseUrl    = env("SHADOW_DATABASE_URL")
}

model User {
  id      Int      @id @default(autoincrement())
  email   String   @unique
  name    String?
  role    Role     @default(USER)
  posts   Post[]
  profile Profile?
}

model Profile {
  id     Int    @id @default(autoincrement())
  bio    String
  user   User   @relation(fields: [userId], references: [id])
  userId Int
}

model Post {
  id         Int      @id @default(autoincrement())
  createdAt  DateTime @default(now())
  uploader   User     @relation(fields: [uploaderId], references: [id])
  uploaderId Int
  title      String
  tags       Tag[]    @relation(references: [id])
  credits    Credit[]
}

model Credit {
  id   Int    @id @default(autoincrement())
  role String
  user String
  link String
}

model Tag {
  id    Int    @id @default(autoincrement())
  name  String
  posts Post[] @relation(references: [id])
}

enum Role {
  USER
  ADMIN
}
```

**Comment by pantharshit00** [comment]
Hello @jdev04 , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

In the meantime, my recommendations is to add a new table which will optionally store the profile and link if profile is not available and in your application you can check if there is a profile linked and display accordingly. 
```prisma
model User {
  id      Int      @id @default(autoincrement())
  email   String   @unique
  name    String?
  role    Role     @default(USER)
  posts   Post[]
  profile Profile?
}

model Profile {
  id            Int               @id @default(autoincrement())
  bio           String
  user          User              @relation(fields: [userId], references: [id])
  userId        Int               @unique
  contributions PostContributor[]
}

model PostContributor {
  id        Int      @id @default(autoincrement())
  post      Post     @relation(fields: [postId], references: [id])
  postId    Int
  profile   Profile? @relation(fields: [profileId], references: [id])
  profileId Int?
  link      String?
}

model Post {
  id               Int               @id @default(autoincrement())
  createdAt        DateTime          @default(now())
  uploader         User              @relation(fields: [uploaderId], references: [id])
  uploaderId       Int
  postContributors PostContributor[]
  title            String
  tags             Tag[]
  credits          Credit[]
}

model Credit {
  id     Int    @id @default(autoincrement())
  role   String
  user   String
  link   String
  Post   Post?  @relation(fields: [postId], references: [id])
  postId Int?
}

model Tag {
  id    Int    @id @default(autoincrement())
  name  String
  posts Post[]
}

enum Role {
  USER
  ADMIN
}
```

**Reply by vladi-stevanovic** [reply]
👋 Hello @jdev04 did you perhaps have a chance to check Harshit's recommendation? 

Let us know if you still have any questions, or feel free to "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
